Objects/Objects/BuiltElements/Revit/DirectShape.cs

```diff
@@ -31,6 +31,16 @@ public DirectShape(string name, RevitCategory category, List<Base> baseGeometrie
     this.parameters = parameters.ToBase();
   }
 
+  // moving away from using the RevitCategory Enum
+  public DirectShape(string name, string builtInCategory, List<Base> baseGeometries, List<Parameter> parameters = null)
+  {
+    this.name = name;
+    this.baseGeometries = baseGeometries.FindAll(IsValidObject);
+    this.parameters = parameters.ToBase();
+    //TODO: move to typed property alongside all other revit elements
+    this["builtInCategory"] = builtInCategory;
```

Comment:
It would be good if we have hard coded strings to move to some constants, especially if referenced in multiple places